crates/biome_js_analyze/src/lint/nursery/use_google_font_preconnect.rs

```diff
@@ -0,0 +1,141 @@
+use biome_analyze::{
+    context::RuleContext, declare_lint_rule, ActionCategory, Ast, FixKind, Rule, RuleDiagnostic,
+    RuleSource, RuleSourceKind,
+};
+use biome_console::markup;
+use biome_js_factory::make;
+use biome_js_syntax::{
+    jsx_ext::AnyJsxElement, AnyJsxAttribute, AnyJsxAttributeName, AnyJsxAttributeValue, T,
+};
+use biome_rowan::{AstNode, AstNodeList, BatchMutationExt, TextRange, TriviaPieceKind};
+
+use crate::JsRuleAction;
+
+declare_lint_rule! {
+    /// Ensure `preconnect` is used with Google Fonts.
+    ///
+    /// When using Google Fonts, adding the `rel="preconnect"` attribute to the `<link>` tag
+    /// that points to `https://fonts.gstatic.com` is recommended to initiate an early
+    /// connection to the font's origin. This improves page load performance by reducing latency.
+    ///
+    /// Failing to use `preconnect` may result in slower font loading times, affecting user experience.
+    ///
+    /// Note: Next.js automatically adds this preconnect link starting from version 12.0.1, but in cases
+    /// where it's manually added, this rule ensures the `preconnect` attribute is properly used.
+    ///
+    /// ## Examples
+    ///
+    /// ### Invalid
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx,expect_diagnostic
+    /// <link rel="preload" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ### Valid
+    ///
+    /// ```jsx
+    /// <link rel="preconnect" href="https://fonts.gstatic.com"/>
+    /// ```
+    ///
+    /// ```jsx
+    /// <link href="/logo.svg" rel="icon" />
+    /// ```
+    ///
+    pub UseGoogleFontPreconnect {
+        version: "next",
+        name: "useGoogleFontPreconnect",
+        language: "jsx",
+        sources: &[RuleSource::EslintNext("google-font-preconnect")],
+        source_kind: RuleSourceKind::SameLogic,
+        recommended: false,
+        fix_kind: FixKind::Safe,
+    }
+}
+
+impl Rule for UseGoogleFontPreconnect {
+    type Query = Ast<AnyJsxElement>;
+    type State = TextRange;
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(ctx: &RuleContext<Self>) -> Self::Signals {
+        let node = ctx.query();
+
+        if node.name().ok()?.name_value_token()?.text_trimmed() != "link" {
+            return None;
+        }
+
+        let href = node.get_attribute_inner_string_text("href")?;
+        let rel = node.get_attribute_inner_string_text("rel");
+
+        if href.starts_with("https://fonts.gstatic.com") && (rel.is_none() || rel? != "preconnect")
```

Comment:
```suggestion
        let rel = node.get_attribute_inner_string_text("rel")?;

        if href.starts_with("https://fonts.gstatic.com") && rel != "preconnect"
```

Since `rel` is mandatory attribute, we can simplify the check

Reply:
It's not mandatory, we emit a diagnostic in the absence of the property or if its value is different of `preconnect`.

Reply:
Then you shouldn't use `rel?`, because the try operator `?` exit from the function in case is `None`, which goes in conflict with the accepted value `rel.is_none`. I suggest using `matches!(rel, Some("preconnect"))` instead